From: George Beshers <gbeshers@redhat.com>
Date: Fri, 5 Feb 2010 16:01:52 -0500
Subject: [pci] update pci_set_vga_state to call arch functions
Message-id: <20100205104608.30315.24756.sendpatchset@sgi-xe250-01.lab.bos.redhat.com>
Patchwork-id: 23143
O-Subject: [RHEL6 PATCH 2/5]
	pci-update-pci_set_vga_state-to-call-arch-functions-fix.patch
Bugzilla: 555879

Except for offsets this is patch
pci-update-pci_set_vga_state-to-call-arch-functions-fix.patch
in the mmotm queue.

From: Andrew Morton <akpm@linux-foundation.org>

Make pci_register_set_vga_state() __init.

Cc: H. Peter Anvin <hpa@zytor.com>
Cc: Ingo Molnar <mingo@elte.hu>
Cc: Jack Steiner <steiner@sgi.com>
Cc: Jesse Barnes <jbarnes@virtuousgeek.org>
Cc: Mike Travis <travis@sgi.com>
Cc: Robin Holt <holt@sgi.com>
Cc: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/drivers/pci/pci.c b/drivers/pci/pci.c
index 9db2180..0a6dae6 100644
--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -2547,7 +2547,7 @@ int pci_resource_bar(struct pci_dev *dev, int resno, enum pci_bar_type *type)
 /* Some architectures require additional programming to enable VGA */
 static arch_set_vga_state_t arch_set_vga_state;
 
-void pci_register_set_vga_state(arch_set_vga_state_t func)
+void __init pci_register_set_vga_state(arch_set_vga_state_t func)
 {
 	arch_set_vga_state = func;	/* NULL disables */
 }
